platform/openide.windows/test/unit/src/org/openide/windows/OnShowingTest.java

```diff
@@ -189,6 +189,21 @@ protected String topComponentID(TopComponent tc, String preferredID) {
                 public TopComponent findTopComponent(String tcID) {
                     throw new UnsupportedOperationException("Not supported yet.");
                 }
+
+                @Override
+                public void createModeFromXml(String xml) {
```

Comment:
A change in test is great! But:

- first of all it signals why your change is incompatible - when you have to update existing the tests, you are doing an incompatible change

- there should be real test showing how to use the new methods that you are introducing.